### ISSUE TITLE
Error while using create_inference_graph - Can't find a device placement for the op!

### ISSUE THREAD

**Issue by srihari-humbarwadi** [tensorflow/tensorflow#21850]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: No
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.10
- **Python version**: 3.5
- **Bazel version (if compiling from source)**:  0.16.1
- **GCC/Compiler version (if compiling from source)**:  5.4.0
- **CUDA/cuDNN version**: 9.0/7.0.5
- **GPU model and memory**: 4 x GTX 1080Ti 394.44 11GB VRAM each
- **TensorRT version**:3.0.4

### Describe the problem
error while running trt.create_inference_graph on a frozen graph pb

### Source code
```
import os
import tensorflow as tf
import tensorflow.contrib.tensorrt as trt
from tensorflow.python.framework import graph_io

_input = 'input_1'
_output = 'output_node0'
outputs = [_output]


def get_frozen_graph():
  with tf.gfile.FastGFile('model.h5.pb', "rb") as f:
    graph_def = tf.GraphDef()
    graph_def.ParseFromString(f.read())
  return graph_def


frozen_graph_def = get_frozen_graph()
trt_graph_def = trt.create_inference_graph(frozen_graph_def, 
					outputs,
					max_batch_size=256, 
					max_workspace_size_bytes=1 << 30, 
					precision_mode='FP16')
tf.reset_default_graph()
g = tf.Graph()
with tf.Session(graph=g) as sess:
	with g.as_default():
		tf.import_graph_def(
  		graph_def=trt_graph_def,
  		name='')
	graph_io.write_graph(g, '.', 'trt_frozen.pb', as_text=False)
```
### logs
2018-08-24 15:34:59.873109: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-08-24 15:34:59.964048: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-08-24 15:35:00.046908: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-08-24 15:35:00.149867: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-08-24 15:35:00.150535: I tensorflow/core/grappler/devices.cc:51] Number of eligible GPUs (core count >= 8): 4
2018-08-24 15:35:00.150615: I tensorflow/core/grappler/clusters/single_machine.cc:359] Starting new session
2018-08-24 15:35:00.151247: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:01:00.0
totalMemory: 10.91GiB freeMemory: 10.38GiB
2018-08-24 15:35:00.151374: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Found device 1 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:02:00.0
totalMemory: 10.92GiB freeMemory: 10.76GiB
2018-08-24 15:35:00.151494: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Found device 2 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:03:00.0
totalMemory: 10.92GiB freeMemory: 10.76GiB
2018-08-24 15:35:00.151608: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Found device 3 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:05:00.0
totalMemory: 10.92GiB freeMemory: 10.76GiB
2018-08-24 15:35:00.156119: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1485] Adding visible gpu devices: 0, 1, 2, 3
2018-08-24 15:35:01.123215: I tensorflow/core/common_runtime/gpu/gpu_device.cc:966] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-08-24 15:35:01.123254: I tensorflow/core/common_runtime/gpu/gpu_device.cc:972]      0 1 2 3 
2018-08-24 15:35:01.123259: I tensorflow/core/common_runtime/gpu/gpu_device.cc:985] 0:   N Y Y Y 
2018-08-24 15:35:01.123263: I tensorflow/core/common_runtime/gpu/gpu_device.cc:985] 1:   Y N Y Y 
2018-08-24 15:35:01.123266: I tensorflow/core/common_runtime/gpu/gpu_device.cc:985] 2:   Y Y N Y 
2018-08-24 15:35:01.123269: I tensorflow/core/common_runtime/gpu/gpu_device.cc:985] 3:   Y Y Y N 
2018-08-24 15:35:01.123691: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1098] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10016 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
2018-08-24 15:35:01.202813: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1098] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 10386 MB memory) -> physical GPU (device: 1, name: GeForce GTX 1080 Ti, pci bus id: 0000:02:00.0, compute capability: 6.1)
2018-08-24 15:35:01.286295: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1098] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:2 with 10386 MB memory) -> physical GPU (device: 2, name: GeForce GTX 1080 Ti, pci bus id: 0000:03:00.0, compute capability: 6.1)
2018-08-24 15:35:01.369518: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1098] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:3 with 10386 MB memory) -> physical GPU (device: 3, name: GeForce GTX 1080 Ti, pci bus id: 0000:05:00.0, compute capability: 6.1)
2018-08-24 15:35:01.827178: I tensorflow/contrib/tensorrt/convert/convert_nodes.cc:2951] Segment @scope '', converted to graph
2018-08-24 15:35:01.827279: E tensorflow/contrib/tensorrt/convert/convert_graph.cc:418] Can't find a device placement for the op!
2018-08-24 15:35:01.901005: W tensorflow/contrib/tensorrt/log/trt_logger.cc:34] DefaultLogger Half2 support requested on hardware without native FP16 support, performance will be negatively affected.
2018-08-24 15:35:02.432465: E tensorflow/contrib/tensorrt/log/trt_logger.cc:38] DefaultLogger resources.cpp (199) - Cuda Error in gieCudaMalloc: 2
2018-08-24 15:35:02.433733: E tensorflow/contrib/tensorrt/log/trt_logger.cc:38] DefaultLogger resources.cpp (199) - Cuda Error in gieCudaMalloc: 2
2018-08-24 15:35:02.434291: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:958] Engine my_trt_op_0 creation for segment 0, composed of 1089 nodes failed: Internal: Failed to build TensorRT engine. Skipping...
2018-08-24 15:35:02.751864: I tensorflow/contrib/tensorrt/convert/convert_nodes.cc:2951] Segment @scope '', converted to graph
2018-08-24 15:35:02.751964: E tensorflow/contrib/tensorrt/convert/convert_graph.cc:418] Can't find a device placement for the op!
2018-08-24 15:35:02.856268: W tensorflow/contrib/tensorrt/log/trt_logger.cc:34] DefaultLogger Half2 support requested on hardware without native FP16 support, performance will be negatively affected.
2018-08-24 15:35:02.904384: E tensorflow/contrib/tensorrt/log/trt_logger.cc:38] DefaultLogger resources.cpp (199) - Cuda Error in gieCudaMalloc: 2
2018-08-24 15:35:02.905595: E tensorflow/contrib/tensorrt/log/trt_logger.cc:38] DefaultLogger resources.cpp (199) - Cuda Error in gieCudaMalloc: 2
2018-08-24 15:35:02.906038: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:958] Engine my_trt_op_0 creation for segment 0, composed of 1089 nodes failed: Internal: Failed to build TensorRT engine. Skipping...
2018-08-24 15:35:03.064587: W tensorflow/contrib/tensorrt/convert/trt_optimization_pass.cc:185] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2018-08-24 15:35:03.127441: W tensorflow/contrib/tensorrt/convert/trt_optimization_pass.cc:185] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2018-08-24 15:35:03.160255: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:403] Optimization results for grappler item: tf_graph
2018-08-24 15:35:03.160298: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   constant folding: Graph size after: 1098 nodes (-606), 1155 edges (-606), time = 96.013ms.
2018-08-24 15:35:03.160304: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   layout: Graph size after: 1161 nodes (63), 1157 edges (2), time = 32.452ms.
2018-08-24 15:35:03.160322: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   TensorRTOptimizer: Graph size after: 1161 nodes (0), 1157 edges (0), time = 808.131ms.
2018-08-24 15:35:03.160329: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   constant folding: Graph size after: 1100 nodes (-61), 1157 edges (0), time = 80.726ms.
2018-08-24 15:35:03.160332: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   TensorRTOptimizer: Graph size after: 1100 nodes (0), 1157 edges (0), time = 390.558ms.
2018-08-24 15:35:03.160335: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:403] Optimization results for grappler item: my_trt_op_0_native_segment
2018-08-24 15:35:03.160349: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   constant folding: Graph size after: 1090 nodes (0), 1147 edges (0), time = 65.266ms.
2018-08-24 15:35:03.160434: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   layout: Invalid argument: The graph is already optimized by layout optimizer.
2018-08-24 15:35:03.160438: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   TensorRTOptimizer: Graph size after: 1090 nodes (0), 1147 edges (0), time = 9.5ms.
2018-08-24 15:35:03.160455: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   constant folding: Graph size after: 1090 nodes (0), 1147 edges (0), time = 53.271ms.
2018-08-24 15:35:03.160458: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:405]   TensorRTOptimizer: Graph size after: 1090 nodes (0), 1147 edges (0), time = 9.467ms.
2018-08-24 15:35:03.336425: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1485] Adding visible gpu devices: 0, 1, 2, 3
2018-08-24 15:35:03.336534: I tensorflow/core/common_runtime/gpu/gpu_device.cc:966] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-08-24 15:35:03.336542: I tensorflow/core/common_runtime/gpu/gpu_device.cc:972]      0 1 2 3 
2018-08-24 15:35:03.336545: I tensorflow/core/common_runtime/gpu/gpu_device.cc:985] 0:   N Y Y Y 
2018-08-24 15:35:03.336548: I tensorflow/core/common_runtime/gpu/gpu_device.cc:985] 1:   Y N Y Y 
2018-08-24 15:35:03.336551: I tensorflow/core/common_runtime/gpu/gpu_device.cc:985] 2:   Y Y N Y 
2018-08-24 15:35:03.336553: I tensorflow/core/common_runtime/gpu/gpu_device.cc:985] 3:   Y Y Y N 
2018-08-24 15:35:03.336822: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1098] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10016 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
2018-08-24 15:35:03.336960: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1098] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 10386 MB memory) -> physical GPU (device: 1, name: GeForce GTX 1080 Ti, pci bus id: 0000:02:00.0, compute capability: 6.1)
2018-08-24 15:35:03.337073: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1098] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:2 with 10386 MB memory) -> physical GPU (device: 2, name: GeForce GTX 1080 Ti, pci bus id: 0000:03:00.0, compute capability: 6.1)
2018-08-24 15:35:03.337142: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1098] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:3 with 10386 MB memory) -> physical GPU (device: 3, name: GeForce GTX 1080 Ti, pci bus id: 0000:05:00.0, compute capability: 6.1)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by qiaohaijun** [comment]
Ｉ use master branch, same error.

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by srihari-humbarwadi** [comment]
Yes it is still an issue

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by benjamintanweihao** [comment]
I'm getting this too. Using TensorRT and TF 1.12.

**Comment by pooyadavoodi** [comment]
This happens when you define a graph while not assigning any device to certain subgraphs that later become TRT ops. This is fine because a TRT op has always a device assigned to it. So, it's safe to ignore this error msg. The message will go away once this PR is merged https://github.com/tensorflow/tensorflow/pull/23866

That being said, I agree that the segmenter needs to be smarter in handling device assignment in ops.

**Comment by srihari-humbarwadi** [comment]
Its present in the latest r1.12.0 (built from source cuda 10 cudnn 7.3.1 nccl 2.3.7 trt 5.0.2)

**Comment by mohantym** [comment]
Hi @srihari-humbarwadi , Could you please refer t[his document](https://docs.nvidia.com/deeplearning/frameworks/tf-trt-user-guide/index.html) for using TF-TRT in latest version.
We  also see that you are using old version of Tensorflow which is officially considered as end of life, We recommend that you upgrade to 2.6 version and let us know if the issue still persists in newer versions .Please open a new issue in case you face any errors, we will get you the right help .Thanks!
